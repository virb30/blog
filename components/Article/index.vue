<template>
    <NuxtLink :to="{ path: `/blog/${props.post.slug}` }">
        <article class="bg-gray-300 dark:bg-gray-600 hover:opacity-80 rounded-lg">
            <header class="h-80 mb-3">
                <img :src="props.post.featureImage" :alt="props.post.slug" class="h-48 mx-auto rounded-t-lg w-full" />
                <h3 class="mt-6 font-bold text-lg px-4 block overflow-ellipsis overflow-hidden">
                    {{ props.post.title }}
                </h3>
                <h5 class="px-4 text-sm opacity-80 mt-2">
                    por {{ props.post.primaryAuthor.name }}
                </h5>
            </header>
            <section class="px-4 mt-1 h-52">
                <p class="text-base overflow-ellipsis overflow-hidden text-justify"
                    :class="[$style['clamp-text'], $style['clamp-lines-lg']]">
                    {{ props.post.excerpt }}
                </p>
            </section>
        </article>
    </NuxtLink>
</template>

<script lang="ts" setup>
import type { PropType } from 'vue';
import Post from '../../domain/entity/Post'

const props = defineProps({
    post: {
        type: Object as PropType<Post>,
        required: true,
    },
})


</script>

<style scoped module lang="postcss">
.clamp-text {
    display: -webkit-box;
    -webkit-line-clamp: 4;
    -webkit-box-orient: vertical;
}

.clamp-lines-sm {
    -webkit-line-clamp: 4;
}

.clamp-lines-lg {
    -webkit-line-clamp: 9;
}
</style>
