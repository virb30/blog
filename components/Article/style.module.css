.clamp-text {
	display: -webkit-box;
	-webkit-line-clamp: 4;
	-webkit-box-orient: vertical;
}

.clamp-lines-sm {
	-webkit-line-clamp: 4;
}

.clamp-lines-lg {
	-webkit-line-clamp: 9;
}
