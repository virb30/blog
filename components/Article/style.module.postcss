article {
	@apply hover:opacity-80  rounded-lg h-80;

	& img {
		@apply h-56 mx-auto rounded-t-lg w-full;
	}

	& div {
		@apply p-4 mt-1;

		& h3 {
			@apply font-bold text-lg;
		}
	}
}
