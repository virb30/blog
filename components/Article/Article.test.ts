import { mount } from '@vue/test-utils';
import Article from './index.vue';

describe('Article Component', () => {
	it('should render post card', () => {
		const wrapper = mount(Article, {
			stubs: {
				NuxtLink: true,
			},
			propsData: {
				post: {
					uuid: 'post-id',
					slug: 'post-slug',
					title: 'Post Title',
					feature_image: '',
					primary_author: {
						name: 'John Doe',
					},
					tags: [
						{ name: 'Teste' },
						{ name: 'Nuxt' },
					],
					updated_at: new Date('2021-10-10'),
				},
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
