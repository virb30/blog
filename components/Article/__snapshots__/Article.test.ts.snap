// Vitest Snapshot v1

exports[`Article Component > should render post card 1`] = `
"<a data-v-91b4acce=\\"\\">
  <article data-v-91b4acce=\\"\\" class=\\"bg-gray-300 dark:bg-gray-600 hover:opacity-80 rounded-lg\\">
    <header data-v-91b4acce=\\"\\" class=\\"h-80 mb-3\\"><img data-v-91b4acce=\\"\\" src=\\"\\" alt=\\"post-slug\\" class=\\"h-48 mx-auto rounded-t-lg w-full\\">
      <h3 data-v-91b4acce=\\"\\" class=\\"mt-6 font-bold text-lg px-4 block overflow-ellipsis overflow-hidden\\">Post Title</h3>
      <h5 data-v-91b4acce=\\"\\" class=\\"px-4 text-sm opacity-80 mt-2\\"> por John Doe</h5>
    </header>
    <section data-v-91b4acce=\\"\\" class=\\"px-4 mt-1 h-52\\">
      <p data-v-91b4acce=\\"\\" class=\\"text-base overflow-ellipsis overflow-hidden text-justify _clamp-text_a58f70 _clamp-lines-lg_a58f70\\">post test</p>
    </section>
  </article>
</a>"
`;
